loopy/codegen/result.py

```diff
@@ -61,6 +61,10 @@ class GeneratedProgram(Record):
         Once generated, this captures the AST of the operative function
         body (including declaration of necessary temporaries), but not
         the overall function definition.
+    .. attribute:: decl_ast
+        Once generated, this captures the AST of the operative function
+        definition but not the function body or declaration of necessary
+        temporaries.
```

Comment:
In a way, there's no reason to keep this around--it could just be recomputed on demand by `generate_header`.